njsscan/__main__.py

```diff
@@ -84,6 +84,65 @@ def json_output(outfile, scan_results):
         return json_output
 
 
+def sonarqube_output(outfile, scan_results):
+    """Sonarqube JSON Output."""
+    sonarqube_issues = []
+    for k, v in scan_results['nodejs'].items():
+        issue = get_sonarqube_issue(v)
+        issue['ruleId'] = k
+        sonarqube_issues.append(issue)
+    sonarqube_report = {
+        'issues': sonarqube_issues,
+    }
+    if outfile:
+        with open(outfile, 'w') as of:
+            json.dump(sonarqube_report, of, sort_keys=True,
+                      indent=2, separators=(',', ': '))
+    else:
+        json_output = (json.dumps(sonarqube_report, sort_keys=True,
+                                  indent=2, separators=(',', ': ')))
+        print(json_output)
+        return json_output
+
+
+def get_sonarqube_issue(njsscan_issue):
+    secondary_locations = []
+    issue_data = njsscan_issue['metadata']
+    for ix, file in enumerate(njsscan_issue['files']):
+        text_range = {
+            'startLine': file['match_lines'][0],
+            'endLine': file['match_lines'][1],
+            'startColumn': file['match_position'][0],
+            'endColumn': file['match_position'][1],
+        }
+        location = {
+            'message': issue_data['description'],
+            'filePath': file['file_path'],
+            'textRange': text_range,
+        }
+        if ix == 0:
+            primary_location = location
+        else:
+            secondary_locations.append(location)
+    issue = {
+        'engineId': 'njsscan',
+        'type': 'VULNERABILITY',
+        'severity': translate_to_sonarqube_severity(issue_data['severity']),
+        'primaryLocation': primary_location,
+    }
+    if secondary_locations:
+        issue['secondaryLocations'] = secondary_locations
+    return issue
+
+
+def translate_to_sonarqube_severity(severity):
```

Comment:
Is the function `translate_to_sonarqube_severity()` required? Can this mapping 
```
d = {
        'ERROR': 'CRITICAL',
        'WARNING': 'MAJOR',
}
```
be inside `get_sonarqube_issue()` or does it make the function complex?